modules/porous_flow/test/tests/jacobian/esbc01.i

```diff
@@ -0,0 +1,99 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+[]
+
+[GlobalParams]
+  PorousFlowDictator = dictator
+  at_nodes = true
+[]
+
+[UserObjects]
+  [./dictator]
+    type = PorousFlowDictator
+    porous_flow_vars = 'pp temp'
+    number_fluid_phases = 1
+    number_fluid_components = 1
+  [../]
+  [./pc]
+    type = PorousFlowCapillaryPressureConst
+    pc = 0.1
+  [../]
+[]
+
+[Variables]
+  [./pp]
+    initial_condition = 1
+  [../]
+  [./temp]
+    initial_condition = 2
+  [../]
+[]
+
+[Kernels]
+  [./mass0]
+    type = TimeDerivative
+    variable = pp
+  [../]
+
+  [./heat_conduction]
+    type = TimeDerivative
+    variable = temp
+  [../]
+[]
+
+[Modules]
+  [./FluidProperties]
+    [./simple_fluid]
+      type = IdealGasFluidProperties
+    [../]
+  [../]
+[]
+
+[Materials]
+  [./ppss]
+    type = PorousFlow1PhaseFullySaturated
+    porepressure = pp
+  [../]
+  [./simple_fluid]
+    type = PorousFlowSingleComponentFluid
+    fp = simple_fluid
+    phase = 0
+  [../]
+
+  [./temperature]
+    type = PorousFlowTemperature
+    temperature = temp
+  [../]
+[]
+
+[BCs]
+  [./left]
+    type = PorousFlowEnthalpySink
+    variable = temp
+    boundary = left
+    pressure = pp
+    T_in = 300
+    fp = simple_fluid
+    flux_function = -23
```

Comment:
This looks like an OK simple test.  It's sometimes quite tricky to devise tests that actually test your code (and not derivatives of enthalpy or whatever).  I assume you've set the test tolerance tight enough to be meaningful.  It would be good if 'flux_function' and/or pp varied across the element, and it would be good if this wasn't a square element, so that you can more convincingly demonstrate you're getting the right jacobian for the right node.